infra/src/main/kotlin/fi/fta/geoviite/infra/inframodel/InfraModelController.kt

```diff
@@ -127,19 +131,40 @@ constructor(
         @RequestParam(name = "lang", defaultValue = "fi") lang: LocalizationLanguage,
     ): ResponseEntity<ByteArray> {
         val translation = localizationService.getLocalization(lang)
+        val localizationParams =
+            getInfraModelBatchFilenameLocalizationParams(
+                applicability,
+                locationTrackId?.let { locationTrackService.get(MainLayoutContext.official, it) },
+                trackNumberId?.let { trackNumberService.get(MainLayoutContext.official, it) },
+                startKmNumber,
+                endKmNumber,
+                translation,
+            )
 
-        return infraModelService.getInfraModelsZipped(ids, translation).let { zipped ->
-            val locationTrackName =
-                locationTrackId?.let { locationTrackService.get(MainLayoutContext.official, it)?.name?.toString() }
-            val trackNumberName =
-                trackNumberId?.let { trackNumberService.get(MainLayoutContext.official, it)?.number.toString() }
-            val alignmentName = requireNotNull(locationTrackName ?: trackNumberName)
-
-            toFileDownloadResponse(
-                getNameForInfraModelZipFile(applicability, alignmentName, startKmNumber, endKmNumber, translation),
-                zipped,
+        val folderName = translation.filename("geometries-folder", LocalizationParams.empty)
+        val summaryFilename =
+            translation.filename(
+                key =
+                    if (startKmNumber != null || endKmNumber != null) "geometry-plans-summary-with-km-range-csv"
+                    else "geometry-plans-summary-csv",
+                params = localizationParams,
             )
-        }
+        val zipFileName =
+            translation.filename(
+                key =
+                    if (startKmNumber != null || endKmNumber != null) "geometry-plans-with-km-range-zip"
+                    else "geometry-plans-zip",
+                params = localizationParams,
+            )
```

Comment:
Tästä endpointista tuli lopulta yllättävän järkäle. Käytännössä kaikki tämä koodi on kuitenkin pelkkää tiedostonimien ja käännösparametrien arpomista ja varsinainen magia tapahtuu tästä eteenpäin